filestuff/mainfiles.java

```diff
@@ -0,0 +1,100 @@
+package filestuff;
+
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Scanner;
+
+import org.apache.poi.hwpf.HWPFDocument;
+import org.apache.poi.hwpf.extractor.WordExtractor;
+import org.apache.poi.xwpf.usermodel.XWPFDocument;
+import org.apache.poi.xwpf.usermodel.XWPFParagraph;
+public class mainfiles {
+
+    //find a file that contains a word in its name inside a directory
+    public static File searchFile(String keyWord, File startfile) {
+        File[] files = startfile.listFiles();
+        for (int i = 0; i < files.length; i++) {
+            if (files[i].isDirectory()) {
+                if (searchFile(keyWord, files[i]) != null) {

```

Comment:
you call the recursive function and if it doesnt return nul you call it again and return it ? sounds wierd